src/ocm/components/clusterConfiguration/networkConfiguration/NetworkConfiguration.tsx

```diff
@@ -0,0 +1,193 @@
+  defaultNetworkSettings: Partial<Cluster>;
+  hideManagedNetworking?: boolean;
+};
+
+const vmsAlert = (
+  <Alert
+    title="Your cluster will be subject to support limitations"
+    variant={AlertVariant.info}
+    isInline={true}
+    data-testid="networking-vms-alert"
+  >
+    Some or all of your discovered hosts are virtual machines, so selecting the cluster-managed
+    networking option will limit your installed cluster's support.
+  </Alert>
+);
+
+const NetworkConfiguration: React.FC<NetworkConfigurationProps> = ({
+  cluster,
+  hostSubnets,
+  isVipDhcpAllocationDisabled,
+  defaultNetworkSettings,
+  hideManagedNetworking,
+  children,
+}) => {
+  const featureSupportLevelData = useFeatureSupportLevel();
+  const { setFieldValue, values, validateField } = useFormikContext<NetworkConfigurationValues>();
+
+  const clusterFeatureSupportLevels = React.useMemo(() => {
+    return getLimitedFeatureSupportLevels(cluster, featureSupportLevelData);
+  }, [cluster, featureSupportLevelData]);
+
+  const isNetworkTypeSelectionEnabled = useFeature(
+    'ASSISTED_INSTALLER_NETWORK_TYPE_SELECTION_FEATURE',
+  );
+
+  const [isAdvanced, setAdvanced] = React.useState(
+    isAdvNetworkConf(cluster, defaultNetworkSettings) || values.stackType === 'dualStack',
+  );
+
+  const isMultiNodeCluster = !isSNO(cluster);
+  const isUserManagedNetworking = values.managedNetworkingType === 'userManaged';
+  const isDualStack = values.stackType === 'dualStack';
+
+  useEffect(() => {
+    if (isUserManagedNetworking) {
+      // We need to reset these fields' values in order to align with the values the server sends
+      setFieldValue('vipDhcpAllocation', false);
+      setFieldValue('ingressVip', '', false);
+      setFieldValue('apiVip', '', false);
+
+      if (isMultiNodeCluster) {
+        setFieldValue('machineNetworks', [], false);
+      }
+    } else {
+      if (!values.vipDhcpAllocation) {
+        validateField('ingressVip');
+        validateField('apiVip');
+      }
+    }
+  }, [
+    isMultiNodeCluster,
+    isUserManagedNetworking,
+    values.vipDhcpAllocation,
+    setFieldValue,
+    validateField,
+  ]);
+
+  const toggleAdvConfiguration = useCallback(
+    (checked: boolean) => {
+      setAdvanced(checked);
+
+      if (!checked) {
+        setFieldValue('clusterNetworks', defaultNetworkSettings.clusterNetworks, false);
+        setFieldValue('serviceNetworks', defaultNetworkSettings.serviceNetworks, false);
+        setFieldValue('networkType', 'OpenShiftSDN');
+      }
+    },
+    [setFieldValue, setAdvanced, defaultNetworkSettings],
+  );
+
+  useEffect(() => {
+    if (values.stackType === 'dualStack' && !isUserManagedNetworking) {
+      toggleAdvConfiguration(true);
+    }
+  }, [values.stackType, toggleAdvConfiguration, isUserManagedNetworking]);
+
+  const enableSDN = useMemo(
+    () =>
+      values.machineNetworks?.every((network) => network.cidr && Address4.isValid(network.cidr)) &&
```

Comment:
This is the place where I say you could reuse the extracted function I named `networkOnlyHasIPXAddresses`